src/Adaptivity/AdaptedTriangulations.jl

```diff
@@ -0,0 +1,316 @@
+  if (get_background_model(strian) === get_background_model(ttrian))
+    return is_change_possible(strian.trian,ttrian.trian)
+  end
+
+  # C) Different background model, but same type of Triangulation (Skeleton, BodyFitted, View, ...)
+  if typeof(strian.trian) == typeof(ttrian.trian) # Is this too restrictive???
+    return is_related(strian,ttrian)
+  end
+
+  # D) Different background model AND different type of triangulation
+  @notimplemented
+  return false
+end
+
+function Geometry.is_change_possible(strian::AdaptedTriangulation,ttrian::Triangulation)
+  # A) Same background model -> Default change of Triangulation
+  if (get_background_model(strian) === get_background_model(ttrian))
+    return is_change_possible(strian.trian,ttrian)
+  end
+  
+  # B) Different background model, but same type of Triangulation (Skeleton, BodyFitted, View, ...)
+  if typeof(strian.trian) == typeof(ttrian)
+    return is_related(strian,ttrian)
+  end
+
+  # C) Different background model AND different type of triangulation
+  @notimplemented
+  return false
+end
+
+function Geometry.is_change_possible(strian::Triangulation,ttrian::AdaptedTriangulation)
+  # A) Same background model -> Default change of Triangulation
+  if (get_background_model(strian) === get_background_model(ttrian))
+    return is_change_possible(strian,ttrian.trian)
+  end
+  
+  # B) Different background model, but same type of Triangulation (Skeleton, BodyFitted, View, ...)
+  if typeof(strian) == typeof(ttrian.trian)
+    return is_related(strian,ttrian)
+  end
+
+  # C) Different background model AND different type of triangulation
+  @notimplemented
+  return false
+end
+
+function Geometry.best_target(strian::AdaptedTriangulation,ttrian::AdaptedTriangulation)
+  @check is_change_possible(strian,ttrian)
+
+  # A) Both Triangulations are exactly the same
+  (strian === ttrian) && (return ttrian)
+
+  # B) Same background model -> Default change of Triangulation
+  if (get_background_model(strian) === get_background_model(ttrian))
+    return best_target(strian.trian,ttrian.trian)
+  end
+
+  # C) Different background model, but same type of Triangulation (Skeleton, BodyFitted, View, ...)
+  if typeof(strian.trian) == typeof(ttrian.trian)
+    is_child(ttrian,strian) ? (return ttrian) : (return strian)
+  end
+
+  # D) Different background model AND different type of triangulation
+  @notimplemented
+  return nothing
+end
+
+function Geometry.best_target(strian::AdaptedTriangulation,ttrian::Triangulation)
+  @check is_change_possible(strian,ttrian)
+  return strian
+end
+
+function Geometry.best_target(strian::Triangulation,ttrian::AdaptedTriangulation)
+  @check is_change_possible(strian,ttrian)
+  return ttrian
+end
+
+function CellData.change_domain(a::CellField,strian::Triangulation,::ReferenceDomain,ttrian::AdaptedTriangulation,::ReferenceDomain)
+  if strian === ttrian
+    return a
+  end
+  @assert is_change_possible(strian,ttrian)
```

Comment:
why `assert` and not `check` ?

Reply:
Addressed in [1d41637](https://github.com/gridap/Gridap.jl/pull/838/commits/1d41637e9b9eee0bce56b6b7ba99c01e435c4eea)